include/solvers/auxiliary_physics.h

```diff
@@ -0,0 +1,149 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2019 - 2019 by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+
+ *
+ * Author: Bruno Blais, Polytechnique Montreal, 2020
+ */
+
+#ifndef lethe_auxiliary_physics_h
+#define lethe_auxiliary_physics_h
+
+#include <deal.II/distributed/tria_base.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <core/parameters.h>
+#include <core/physics_solver.h>
+#include <solvers/simulation_parameters.h>
+
+
+/**
+ * Au auxiliary physics is defined as a physics that is solved on top of
+ * a core physics, the latter being the Navier-Stokes
+ * equations. Auxiliary physics are simpler
```

Comment:
> Very cool new architecture! It clarifies a lot the code and it will be very handy in the upcoming developments.

Let's hope so! It has less limitations than I actually had imagined so it's good to see something better than we thought.